### ISSUE TITLE
SequentialLR changes the schedule of nested composite schedulers

### ISSUE THREAD

**Issue by pupeno** [pytorch/pytorch#196243]
### 🐛 Describe the bug

Adding another `SequentialLR` around a composite learning rate scheduler changes the schedule it produces.

The outer `SequentialLR` in this example has only one scheduler and no milestones, so it should behave as a transparent wrapper. Instead, constructing it advances the wrapped scheduler by one step:

```python
import torch
from torch.optim import SGD
from torch.optim.lr_scheduler import (
    ChainedScheduler,
    ConstantLR,
    ExponentialLR,
    SequentialLR,
)


def get_lrs(scheduler):
    lrs = []
    for _ in range(5):
        lrs.append(round(scheduler.get_last_lr()[0], 6))
        scheduler.optimizer.step()
        scheduler.step()
    return lrs


def make_sequential(optimizer):
    return SequentialLR(
        optimizer,
        [
            ConstantLR(optimizer, factor=0.5, total_iters=2),
            ConstantLR(optimizer, factor=0.2, total_iters=10),
        ],
        milestones=[2],
    )


two_level_optimizer = SGD([torch.nn.Parameter(torch.zeros(1))], lr=0.1)
two_level_scheduler = SequentialLR(
    two_level_optimizer,
    [make_sequential(two_level_optimizer)],
    milestones=[],
)

one_level_optimizer = SGD([torch.nn.Parameter(torch.zeros(1))], lr=0.1)
one_level_scheduler = make_sequential(one_level_optimizer)

print("SequentialLR")
print("    one level: ", get_lrs(one_level_scheduler))
print("    two levels:", get_lrs(two_level_scheduler))


def make_chained(optimizer):
    return ChainedScheduler(
        [
            ConstantLR(optimizer, factor=0.5, total_iters=2),
            ExponentialLR(optimizer, gamma=0.9),
        ],
        optimizer=optimizer,
    )


two_level_optimizer = SGD([torch.nn.Parameter(torch.zeros(1))], lr=0.1)
two_level_scheduler = SequentialLR(
    two_level_optimizer,
    [make_chained(two_level_optimizer)],
    milestones=[],
)

one_level_optimizer = SGD([torch.nn.Parameter(torch.zeros(1))], lr=0.1)
one_level_scheduler = make_chained(one_level_optimizer)

print("ChainedScheduler")
print("    one level: ", get_lrs(one_level_scheduler))
print("    two levels:", get_lrs(two_level_scheduler))
```

Ignoring the warnings described below, the output is:

```
SequentialLR
    one level:  [0.05, 0.05, 0.02, 0.02, 0.02]
    two levels: [0.05, 0.02, 0.02, 0.02, 0.02]
ChainedScheduler
    one level:  [0.05, 0.045, 0.081, 0.0729, 0.06561]
    two levels: [0.045, 0.0405, 0.0729, 0.06561, 0.059049]
```

The two-level `SequentialLR` reaches its milestone one user step early, so the second `ConstantLR` starts at epoch 1 instead of epoch 2.

The two-level `ChainedScheduler` performs a normal chained step while the outer scheduler is being constructed. In this example, that applies `ExponentialLR`'s decay at epoch 0, before the user has called `step()`.

Both cases also emit the warning that `lr_scheduler.step()` was called before `optimizer.step()`, even though the user has not called either method yet.

### Cause

`SequentialLR.__init__` activates its first scheduler by calling `scheduler._initial_step()`. `SequentialLR` and `ChainedScheduler` do not override `_initial_step`, so they inherit `LRScheduler._initial_step`, which calls `self.step()`.

Calling `step()` is the correct initialization for a leaf scheduler. For a composite scheduler, however, it advances the schedule that its child schedulers produce.

`SequentialLR.recursive_undo()` rewinds the initialization steps taken by the leaf schedulers before the active scheduler is initialized. It cannot compensate for this extra composite step because it only adjusts the leaf schedulers' `last_epoch` values.

### Expected behavior

The one-level and two-level schedulers should produce identical learning rates. Adding a single-child `SequentialLR` with no milestones should not advance the wrapped scheduler or emit a step-order warning.

### Versions

PyTorch version: 2.15.0a0+git24bcbe1
Is debug build: False
CUDA used to build PyTorch: 13.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 26.04.1 LTS (x86_64)
GCC version: (Ubuntu 15.2.0-16ubuntu1) 15.2.0
Clang version: Could not collect
CMake version: version 4.4.2
Libc version: glibc-2.43

Python version: 3.14.4 (main, Jun 18 2026, 14:25:02) [GCC 15.2.0] (64-bit runtime)
Python platform: Linux-7.0.0-31-generic-x86_64-with-glibc2.43
Is CUDA available: True
CUDA runtime version: 13.2.86
CUDA_MODULE_LOADING set to:
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3050

Nvidia driver version: 595.84
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_tensor_ir.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_ext.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.25.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.25.1
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           39 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  32
On-line CPU(s) list:                     0-31
Vendor ID:                               GenuineIntel
Model name:                              13th Gen Intel(R) Core(TM) i9-13900KF
CPU family:                              6
Model:                                   183
Thread(s) per core:                      2
Core(s) per socket:                      24
Socket(s):                               1
Stepping:                                1
CPU(s) scaling MHz:                      47%
CPU max MHz:                             5800.0000
CPU min MHz:                             800.0000
BogoMIPS:                                5990.40
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                          VT-x
L1d cache:                               896 KiB (24 instances)
L1i cache:                               1.3 MiB (24 instances)
L2 cache:                                32 MiB (12 instances)
L3 cache:                                36 MiB (1 instance)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-31
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Mitigation; Clear Register File
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] numpy==2.5.2
[pip3] optree==0.20.0
[pip3] torch==2.15.0a0+git24bcbe1
[conda] Could not collect
